### PR TITLE
Export SmokescreenContext type

### PR DESCRIPTION
**Pull request by cmoresco-stripe**
In https://github.com/stripe/smokescreen/pull/199 we added a hook that will be called on a CONNECT response and takes a smokescreenContext. In order for callers to be able to write functions that use smokescreenContext, we need to export the type.

### CODE REVIEW

File: pkg/smokescreen/smokescreen.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ type aclDecision struct {
 	enforceWouldDeny                    bool
 }
 
-type smokescreenContext struct {
+type SmokescreenContext struct {
 	cfg           *Config
 	start         time.Time
 	decision      *aclDecision
```
Comment (jjiang-stripe) [inline pkg/smokescreen/smokescreen.go:76]: Do we also need to export this field?

**Review by jjiang-stripe — CHANGES_REQUESTED** [review]

File: pkg/smokescreen/smokescreen.go (hunk 1 above)
Comment (cmoresco-stripe) [reply pkg/smokescreen/smokescreen.go]: Good call! Also ResolvedAddr. Thanks!

File: pkg/smokescreen/smokescreen.go [hunk 2]
```diff
@@ -63,17 +63,17 @@ const (
...
 	allow                               bool
 	enforceWouldDeny                    bool
 }
 
-type smokescreenContext struct {
+type SmokescreenContext struct {
 	cfg           *Config
 	start         time.Time
-	decision      *aclDecision
+	decision      *ACLDecision
```
Comment (jjiang-stripe) [inline pkg/smokescreen/smokescreen.go:76]:
```suggestion
	Decision      *ACLDecision
```

**Review by jjiang-stripe — CHANGES_REQUESTED** [review]
final lil capitalization bit

**Review by jjiang-stripe — APPROVED** [review]

**Pull request merged**